Remove deprication code for reolink Hub switches

dollaransh17 opened the pull request:
Fixed a typo in the variable name `depricated_dict` which should be `deprecated_dict`. 

This is a simple spelling correction in 4 places within the reolink switch component code. 

No functional changes - just correcting the spelling for better code readability.
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change

<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.

-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [x] I understand the code I am submitting and can explain how it works.
- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.
- [x] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

dollaransh17 committed:
Fix typo: 'depricated' to 'deprecated' in reolink switch component

dollaransh17 committed:
Merge branch 'dev' into fix-typo-deprecated-reolink

joostlek reviewed (dismissed):
Why don't we remove this deprecation at all? It has been overdue for 6 months already

starkillerOG reviewed (approved):
LGTM

dollaransh17 committed:
Remove overdue deprecated NVR switch entities

As requested by maintainers, removing the deprecated switch entities
that were scheduled for removal in HA 2025.4.0 (6 months overdue).

This removes:
- DEPRECATED_NVR_SWITCHES definition
- deprecated_dict handling code
- hub_switch_deprecated issue creation

Also fixes typo: 'depricated' → 'deprecated' in variable names.

dollaransh17 committed:
Merge branch 'dev' into fix-typo-deprecated-reolink

dollaransh17 commented:
> Yes, let's go ahead and remove the Deprecation code in this PR

plz have a look sir , and plz merge it

starkillerOG commented on homeassistant/components/reolink/switch.py:
```diff
@@ -389,34 +339,6 @@ async def async_setup_entry(
-                "hub_switch_deprecated",
-                is_fixable=False,
-                severity=ir.IssueSeverity.WARNING,
-                translation_key="hub_switch_deprecated",
```
Please also remove this translation from the strings.json

starkillerOG commented:
And please make sure all tests are passing (Ruff styling)

dollaransh17 committed:
Remove unused imports and deprecated translation

- Remove unused entity_registry and issue_registry imports
- Remove hub_switch_deprecated translation from strings.json

These were only used by the deprecated code that was removed.

dollaransh17 commented:
> And please make sure all tests are passing (Ruff styling)

Sir plz see it and take the further steps...

starkillerOG commented:
You have to remove `from .const import DOMAIN`

dollaransh17 committed:
Remove unused DOMAIN import

DOMAIN was only used in the deprecated ir.async_create_issue() call.

starkillerOG commented:
Tests are failing.
You will need to cleanup the `test_hub_switches_repair_issue` and `test_cleanup_hub_switches` tests from `tests/components/reolink/test_switch.py`

Since the code which those tests are testing is now removed, those tests can also be safely removed.

dollaransh17 committed:
Remove tests for deprecated hub switches

Remove test_cleanup_hub_switches and test_hub_switches_repair_issue.

dollaransh17 committed:
Remove unused imports from test_switch.py

dollaransh17 committed:
Remove DOMAIN and TEST_UID unused imports from test.

starkillerOG commented:
Thank you, good work!

dollaransh17 commented:
> Thank you, good work!

Thank you sir.
